src/main/java/com/jwplayer/southpaw/Southpaw.java

```diff
@@ -225,165 +248,56 @@ public Southpaw(Map<String, Object> rawConfig, Relation[] relations)
             byte[] bytes = state.get(METADATA_KEYSPACE, createDePKEntryName(root).getBytes());
             dePKsByType.put(root, ByteArraySet.deserialize(bytes));
         }
+
+        // Start the global timers
+        backupWatch.start();
+        commitWatch.start();
+        metricsWatch.start();
     }
 
     /**
      * Reads batches of new records from each of the input topics and creates the appropriate denormalized
      * records according to the top level relations. Performs a full commit and backup before returning.
-     * @param runTimeS - Sets an amount of time in seconds for this method to run. The method will not run this amount
-     *                of time exactly, but will stop after processing the latest batch of records. If set to 0,
-     *                it will run until interrupted. Probably most useful for testing.
+     * @param runTimeMS - Sets an amount of time in milliseconds for this method to run. The method will not run this
+     *                amount of time exactly, but will stop after processing the latest batch of records. If set to 0,
+     *                it will run until interrupted. Useful for testing.
      */
-    protected void build(int runTimeS) {
+    protected void build(long runTimeMS) {
         logger.info("Building denormalized records");
-        StopWatch backupWatch = new StopWatch();
-        backupWatch.start();
         StopWatch runWatch = new StopWatch();
         runWatch.start();
-        StopWatch commitWatch = new StopWatch();
-        commitWatch.start();
 
         List<Map.Entry<String, BaseTopic<BaseRecord, BaseRecord>>> topics = new ArrayList<>(inputTopics.entrySet());
         List<String> rootEntities = Arrays.stream(relations).map(Relation::getEntity).collect(Collectors.toList());
         topics.sort((x, y) -> Boolean.compare(rootEntities.contains(x.getKey()), rootEntities.contains(y.getKey())));
 
         while(processRecords) {
-            // Loop through each input topic and read a batch of records
-
             for (Map.Entry<String, BaseTopic<BaseRecord, BaseRecord>> entry : topics) {
-                String entity = entry.getKey();
                 BaseTopic<BaseRecord, BaseRecord> inputTopic = entry.getValue();
-
-                long topicLag;
-                calculateRecordsToCreate();
-                calculateTotalLag();
-
                 do {
+                    String entity = entry.getKey();
                     Iterator<ConsumerRecord<BaseRecord, BaseRecord>> records = inputTopic.readNext();
-                    // Loop through each record and process it
                     while (records.hasNext()) {
-                        ConsumerRecord<BaseRecord, BaseRecord> newRecord = records.next();
-                        ByteArray primaryKey = newRecord.key().toByteArray();
+                        ConsumerRecord<BaseRecord, BaseRecord> inputRecord = records.next();
                         for (Relation root : relations) {
```

Comment:
Do you think it's worth removing the logic for supporting multiple relations? We haven't used this in years. Could simplify the loop a bit more

Reply:
Eh, I don't think it's worth it to add even more commits on top of this already lengthy PR. And theoretically someone out there is using Southpaw with multiple relations in a single instance.